### PR TITLE
New error logging

### PR DESCRIPTION
**Pull request by Astroner**

### CODE REVIEW

File: src/parser/getNodesToTranspile.ts [hunk 1]
```diff
@@ -37,21 +43,34 @@ export const getNodesToTranspile = (
                 const checked = checker.getTypeAtLocation(node);
 
                 // check that type is a shape declaration
-                if (!checked.symbol?.declarations?.length) continue;
+                if (!checked.symbol?.declarations?.length) {
+                    console.warn(
```
Comment (ipavelleds) [inline src/parser/getNodesToTranspile.ts:47]: Could we show these warnings at the end of logs? 'cause we could easily miss these warnings

File: src/utils/joinLeft.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export const joinLeft = <LL, LR, R>(
+    either: Either<Either<LL, LR>, R>
+): Either<LL | LR, R> => {
+    if (isRight(either)) return either;
+    const leftEither = either.left;
+
+    if (isRight(leftEither)) return left(leftEither.right);
+
+    return leftEither;
```
Comment (ipavelleds) [inline src/utils/joinLeft.ts]: could be simplified
```typescript
pipe(
        either,
        mapLeft(getOrElseW(identity))
    )
```

File: src/utils/sequenceReaderEither.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        } else {
+            values.push(either.right);
+        }
+    });
+
+    if (errors.length > 0) {
+        return left(errors);
+    }
+
+    return right(values);
```
Comment (ipavelleds) [inline src/utils/sequenceReaderEither.ts]: can be simplified
```typescript
pipe(
    items,
    partitionMap(r => r(env)),
    ({ left, right}) => (
        pipe(
            right,
            eitherFromPredicate(constant(array.isEmpty(left)), constant(left))
        )
    )
)
```

**Review by ipavelleds — CHANGES_REQUESTED** [review]

File: src/parser/getNodesToTranspile.ts (hunk 1 above)
Comment (Astroner) [reply src/parser/getNodesToTranspile.ts:47]: I think no, we can't coz for this improvement I'll have to refactor whole the code(replace Either with These)

**Review by ipavelleds — APPROVED** [review]

**Pull request merged**